src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.SymbolToProjectId.cs

```diff
@@ -51,9 +53,24 @@ internal partial class SolutionState
             {
                 if (!unrootedSymbolToProjectId.TryGetValue(symbol, out var projectId))
                 {
+                    // look through all the projects, and see if this symbol came from the primary assembly for that
+                    // project.  (i.e. was a source symbol from that project, or was retargetted into that project).  If
+                    // so, that's the originating project.
                     foreach (var (id, tracker) in _projectIdToTrackerMap)
                     {
-                        if (tracker.ContainsAssemblyOrModuleOrDynamic(symbol))
+                        if (tracker.ContainsAssemblyOrModuleOrDynamic(symbol, primary: true))
+                        {
+                            projectId = id;
+                            break;
+                        }
+                    }
+
+                    // now, look through all the secondary assemblies (i.e. references) for that project.  This is the
+                    // case for metadata symbols.  A metadata symbol might be found in many projects, so we just return
+                    // the first result as that's just as good for finding the metadata symbol as any other project.
+                    foreach (var (id, tracker) in _projectIdToTrackerMap)
```

Comment:
❓  Did you mean to run through this loop even when `projectId` was set for the previous one?

Reply:
📝 based on one of your PR comments, it sounds like this may be a bug in the current code.

Reply:
good catch.

Reply:
fixed.